src/support/dump-collection.ts

```diff
@@ -3,86 +3,234 @@ import path from 'path'
 
 import cli from 'cli-ux'
 import ndjson from 'ndjson'
+import { pickValues } from '../common'
+
+abstract class AbstractDumper {
+  protected collectionDir: string
+  protected filename?: string
+  protected writeStream?: fs.WriteStream
+  protected options: {scroll: string, size: number}
+
+  protected abstract get fileExtension(): string
+
+  constructor(
+    protected readonly sdk: any,
+    protected readonly index: string,
+    protected readonly collection: string,
+    protected readonly batchSize: number,
+    protected readonly destPath: string,
+    protected readonly query: any = {},
+  ) {
+    this.collectionDir = path.join(this.destPath, this.collection)
+    this.options = {
+      scroll: '1m',
+      size: batchSize
+    }
+  }
 
-export async function dumpCollectionData(sdk: any, index: string, collection: string, batchSize: number, destPath: string, query: any = {}, format = 'jsonl') {
-  const collectionDir = path.join(destPath, collection)
-  const filename = path.join(collectionDir, (format.toLowerCase() === 'jsonl' ? 'documents.jsonl' : 'documents.json'))
-  const writeStream = fs.createWriteStream(filename)
-  const waitWrite = new Promise(resolve => writeStream.on('finish', resolve))
-  const ndjsonStream = ndjson.stringify()
-  const options = {
-    scroll: '1m',
-    size: batchSize
+  /**
+   * One-shot call before the dump. Can be used to
+   * perform setup operations before dumping.
+   *
+   * @returns void
```

Comment:
Nitpicking: there is little interest to document the void return since the TS type will already indicate this